url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/cmdlib/MathLibrary.java

```diff
@@ -0,0 +1,197 @@
+import com.sdicons.scripty.parser.IContext;
+import com.sdicons.scripty.spec.type.BigDecimalType;
+import com.sdicons.scripty.spec.type.TypeSpecException;
+import com.sdicons.scripty.annot.*;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+
+@ScriptyNamedArgLists(
+        std={
+                @ScriptyStdArgList(name="2numbers", fixed={@ScriptyArg(name="arg1", type="BigDecimal"), @ScriptyArg(name="arg2", type="BigDecimal")}),
+                @ScriptyStdArgList(name="1object", fixed={@ScriptyArg(name="arg", type="Any")}),
+                @ScriptyStdArgList(name="1number", fixed={@ScriptyArg(name="arg", type="BigDecimal")})
+        },
+        var={
+                @ScriptyVarArgList(name="numbers*", fixed={}, named={}, vararg=@ScriptyArg(name="numbers", type="BigDecimal")),
+                @ScriptyVarArgList(name="numbers+", fixed={}, named={}, vararg=@ScriptyArg(name="numbers", type="BigDecimal"), minLength = 1)}
+)
+@ScriptyLibrary(type=ScriptyLibraryType.AUTO)
+public class MathLibrary
+{
+    private MathContext mathCtx = MathContext.DECIMAL64;
+    
+    @ScriptyCommand(name="+")
+    @ScriptyRefArgList(ref = "numbers*")
+    public BigDecimal add(@ScriptyParam("numbers") Object[] aNumbers)
+    {
+        BigDecimal lSum = BigDecimal.ZERO;
+        for(Object lDec : aNumbers)
+        {
+            lSum = lSum.add((BigDecimal) lDec, mathCtx);
+        }
+        return lSum;
+    }
+
+    @ScriptyCommand(name="-")
+    @ScriptyRefArgList(ref = "numbers+")
+    public BigDecimal sub(@ScriptyParam("numbers") Object[] aNumbers)
+    {
+        BigDecimal lSum = (BigDecimal) aNumbers[0];
+        for(int i = 1; i < aNumbers.length; i++)
+        {
+            lSum = lSum.subtract((BigDecimal) aNumbers[i], mathCtx);
+        }
+        return lSum;
+    }
+
+    @ScriptyCommand(name="/")
+    @ScriptyRefArgList(ref = "numbers+")
+    public BigDecimal div(@ScriptyParam("numbers") Object[] aNumbers)
+    {
+        BigDecimal lResult = (BigDecimal) aNumbers[0];
+        for(int i = 1; i < aNumbers.length; i++)
+        {
+            lResult = lResult.divide((BigDecimal)aNumbers[i],mathCtx);
+        }
+        return lResult;
+    }
+
+    @ScriptyCommand(name="*")
+    @ScriptyRefArgList(ref = "numbers+")
+    public BigDecimal mult(@ScriptyParam("numbers") Object[] aNumbers)
+    {
+        BigDecimal lResult = (BigDecimal) aNumbers[0];
+        for(int i = 1; i < aNumbers.length; i++)
+        {
+            lResult = lResult.multiply((BigDecimal) aNumbers[i], mathCtx);
+        }
+        return lResult;
+    }
+
+    @ScriptyCommand(name="^")
+    @ScriptyRefArgList(ref="2numbers")
+    public BigDecimal pow(@ScriptyParam("arg1") BigDecimal aNr, @ScriptyParam("arg2") BigDecimal aPow)
+    throws CommandException
+    {
+        try
+        {
+            return aNr.pow(aPow.intValueExact(),mathCtx);
+        }
+        catch (ArithmeticException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)